src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    let stateCopy = { ...currentState };
```

Comment:
You create a copy of the current state at the start of each iteration. 
To reduce code duplication, we do this after processing the action.
```suggestion
```
``result.push({ ...currentState });``